homeassistant/components/zwave_js/diagnostics.py

```diff
@@ -34,24 +34,32 @@
 )
 
 
-def redact_value_of_zwave_value(zwave_value: ValueDataType) -> ValueDataType:
-    """Redact value of a Z-Wave value."""
+def _redacted_value(zwave_value: ValueDataType) -> ValueDataType:
+    """Return redacted value of a Z-Wave value."""
+    redacted_value: ValueDataType = deepcopy(zwave_value)
+    redacted_value["value"] = REDACTED
+    return redacted_value
+
+
+def optionally_redact_value_of_zwave_value(zwave_value: ValueDataType) -> ValueDataType:
+    """Redact value of a Z-Wave value if it matches criteria to redact."""
     # If the value has no value, there is nothing to redact
     if zwave_value.get("value") in (None, ""):
         return zwave_value
+    if zwave_value.get("metadata", {}).get("secret"):
+        return _redacted_value(zwave_value)
```

Comment:
Can't we just use `async_redact_data` to always redact `secret`?

Reply:
We aren't redacting the secret key, we are redacting the value key if the secret key is true and it's nested within the dict. I don't think async_redact_data would work for this case.

Reply:
Thanks 👍